doc/neps/nep-0034.rst

```diff
@@ -76,41 +78,43 @@ indicate failure.
 Backward compatibility
 ----------------------
 
-Anyone depending on ragged lists-of-lists creating object arrays will need to
+Anyone depending on ragged nested sequences creating object arrays will need to
 modify their code. There will be a deprecation period during which the current
 behaviour will emit a ``DeprecationWarning``.
 
 
 Alternatives
 ------------
 
-We could continue with the current situation.
+- We could continue with the current situation.
 
-It was also suggested to add a kwarg `depth` to array creation, or perhaps to
-add another array creation API function `ragged_array_object`. The goal was
-to eliminate the ambiguity in creating an object array from `array([[1, 2],
-[1]], dtype=object)`: should the returned array have a shape of `(1,)`, or
-`(2,)`? This NEP does not deal with that issue, and only deprecates the use of
-`array` with no `dtype=object` for ragged arrays.
+- It was also suggested to add a kwarg ``depth`` to array creation, or perhaps
+  to add another array creation API function ``ragged_array_object``. The goal
+  was to eliminate the ambiguity in creating an object array from ``array([[1,
+  2], [1]], dtype=object)``: should the returned array have a shape of
+  ``(1,)``, or ``(2,)``? This NEP does not deal with that issue, and only
+  deprecates the use of ``array`` with no ``dtype=object`` for ragged nested
+  sequences. Users of ragged nested sequences may face another deprecation
+  cycle in the future.
 
-It was also suggested to deprecate all automatic creation of ``object``-dtype
-arrays, which would require a dtype for something like ``np.array([Decimal(10),
-Decimal(10)])``. This too is out of scope for the current NEP: only if all the
-top-level elements are `sequences`_ will we require an explicit
-``dtype=object``.
+- It was also suggested to deprecate all automatic creation of ``object``-dtype
+  arrays, which would require a dtype for something like ``np.array([Decimal(10),
+  Decimal(10)])``. This too is out of scope for the current NEP: only if all
+  the top-level elements are `sequences`_ will we require an explicit
+  ``dtype=object``.
```

Comment:
This is only if they are ragged, right? `np.array([[[Decimal(1)]]])` is still fine?

Reply:
the intention is that should still just work

Reply:
In that case, I think this sentence needs rewording

Reply:
removed the confusing last clause of the sentence